misk/web/tabs/loader/src/actions/types.ts

```diff
@@ -0,0 +1,22 @@
+import { createActionTypes } from "./utils"
+
+export interface IActionType {
+  [base:string]: string
+}
+
+export const ITEM = createActionTypes("ITEM", [
+  "GET",
+  "GET_ONE",
+  "SAVE",
+  "PUT",
+  "PATCH",
+  "DELETE",
+  "SUCCESS",
+  "FAILURE"
+])
+
+export const ADMINTABS = createActionTypes("ADMINTABS", [
+  "FAILURE",
+  "GET_ALL",
+  "SUCCESS"
+])
```

Comment:
Are these different from `enum`s? A basic reading of this leads me to believe you won't get any type-checking guarantees with this setup.

Reply:
Your reading is correct, the types are not type checked currently. I've created an issue to come back to that includes some initial research on how best to do this as there doesn't seem to be an immediately default or intuitive approach. #367